File: NetbeansProjects/AStar/src/test/TestNode.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package test;

import astar.Edge;
import astar.Node;
import java.util.ArrayList;
import java.util.Iterator;

/**
 *
 * @author Tobias
 */
public class TestNode implements Node<TestNode>
{
    private final int x;
    private final int y;
    private final int moveCost;
    private final ArrayList<Edge<TestNode>> edges;

    public TestNode(int x, int y, int moveCost)
    {
        this.x = x;
        this.y = y;
        this.moveCost = moveCost;
        this.edges = new ArrayList<>();
    }

    @Override
    public int hashCode()
    {
        int hash = 7;
        hash = 37 * hash + this.x;
        hash = 37 * hash + this.y;
        return hash;
    }

    @Override
    public boolean equals(Object obj)
    {
        if (this == obj)
        {
            return true;
        }
        if (obj == null)
        {
            return false;
        }
        if (getClass() != obj.getClass())
        {
            return false;
        }
        final TestNode other = (TestNode) obj;
        if (this.x != other.x)
        {
            return false;
        }
        return this.y == other.y;
    }
    
    

    public int getX()
    {
        return x;
    }

    public int getY()
    {
        return y;
    }

    public int getMoveCost()
    {
        return moveCost;
    }

    public void addEdge(TestNode node)
    {
        Edge e = new TestEdge(node, this.moveCost + node.moveCost); 
        edges.add(e);
    }

    @Override
    public Iterator<Edge<TestNode>> iterator()
    {
        return edges.iterator();
    }

    @Override
    public String toString()
    {
        return "(" + x + ", " + y + ") " + moveCost;
    }

    private static class TestEdge implements Edge<TestNode>
    {

        private TestNode end;
        private float cost;

        public TestEdge(TestNode end, float cost)
        {
            this.end = end;
            this.cost = cost;
        }

        @Override
        public float getCost()
        {
            return cost;
        }

        @Override
        public TestNode getEnd()
        {
            return end;
        }

    }

}

File: VisualStudioProjects/GraphAlgorithms/GraphAlgorithms/TestNode.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Text;

namespace GraphAlgorithms.Test
{
    class TestNode : INode<TestNode>
    {
        public int X { get; }
        public int Y { get; }
        public int MoveCost { get; }
        private ICollection<IEdge<TestNode>> edges;

        public override string ToString()
        {
            return "(" + X + ", " + Y + ") :" + MoveCost;
        }

        public TestNode(int x, int y, int moveCost)
        {
            this.X = x;
            this.Y = y;
            this.MoveCost = moveCost;
            this.edges = new List<IEdge<TestNode>>();
        }

        public void AddEdgeTo(TestNode node)
        {
            IEdge<TestNode> edge = new TestEdge(MoveCost + node.MoveCost, node);
            edges.Add(edge);
        }

        public override bool Equals(object obj)
        {
            var node = obj as TestNode;
            return node != null &&
                   X == node.X &&
                   Y == node.Y;
        }

        public IEnumerator<IEdge<TestNode>> GetEnumerator()
        {
            return edges.GetEnumerator();
        }

        public override int GetHashCode()
        {
            var hashCode = 1502939027;
            hashCode = hashCode * -1521134295 + X.GetHashCode();
            hashCode = hashCode * -1521134295 + Y.GetHashCode();
            return hashCode;
        }

        IEnumerator IEnumerable.GetEnumerator()
        {
            return edges.GetEnumerator();
        }



        private class TestEdge : IEdge<TestNode>
        {
            private readonly float cost;
            private TestNode end;

            public TestEdge(float cost, TestNode end)
            {
                this.cost = cost;
                this.end = end;
            }

            public float GetCost()
            {
                return cost;
            }

            public TestNode GetEnd()
            {
                return end;
            }
        }
    }
}
